frame/transaction-payment/rpc/src/lib.rs

```diff
@@ -69,19 +69,24 @@ impl From<Error> for i64 {
 	}
 }
 
-impl<C, Block, Balance> TransactionPaymentApi<<Block as BlockT>::Hash, RuntimeDispatchInfo<Balance>>
+impl<C, Block, Weight, DispatchClass, Balance> TransactionPaymentApi<
+	<Block as BlockT>::Hash,
+	RuntimeDispatchInfo<Weight, DispatchClass, Balance>,
+>
 	for TransactionPayment<C, Block>
 where
 	Block: BlockT,
 	C: Send + Sync + 'static + ProvideRuntimeApi<Block> + HeaderBackend<Block>,
-	C::Api: TransactionPaymentRuntimeApi<Block, Balance>,
+	C::Api: TransactionPaymentRuntimeApi<Block, Weight, DispatchClass, Balance>,
+	Weight: Codec,
+	DispatchClass: Codec,
 	Balance: Codec + MaybeDisplay + MaybeFromStr,
 {
 	fn query_info(
 		&self,
 		encoded_xt: Bytes,
 		at: Option<<Block as BlockT>::Hash>
-	) -> Result<RuntimeDispatchInfo<Balance>> {
+	) -> Result<RuntimeDispatchInfo<Weight, DispatchClass, Balance>> {
```

Comment:
We sort of need to return Psudo:`RuntimeDispatchInfo<<Block as BlockT>::Weight, <Block as BlockT>::DispatchClass, <Block as BlockT>::Balance>`

Reply:
Hmm, it may not be right this change I am doing then because these are not associated types which we can grab from `Block`... nor does it really make sense to change it that way.